vault-github/run.sh

```diff
@@ -31,57 +31,68 @@ unzip() {
+      printf "Failed to determine the latest Vault version.\n"
+      return 1
+    fi
+    INSTALL_VERSION=$${LATEST_VERSION}
   fi
-  VERSION=$${LATEST_VERSION}
-fi
 
-# Check if the vault CLI is installed and has the correct version
-installation_needed=1
-if command -v vault > /dev/null 2>&1; then
-  CURRENT_VERSION=$(vault version | grep -oE '[0-9]+\.[0-9]+\.[0-9]+')
-  if [ "$${CURRENT_VERSION}" = "$${INSTALL_VERSION}" ]; then
-    printf "Vault version %s is already installed and up-to-date.\n\n" "$${CURRENT_VERSION}"
-    installation_needed=0
+  # Check if the vault CLI is installed and has the correct version
+  installation_needed=1
+  if command -v vault > /dev/null 2>&1; then
+    CURRENT_VERSION=$(vault version | grep -oE '[0-9]+\.[0-9]+\.[0-9]+')
+    if [ "$${CURRENT_VERSION}" = "$${INSTALL_VERSION}" ]; then
+      printf "Vault version %s is already installed and up-to-date.\n\n" "$${CURRENT_VERSION}"
+      installation_needed=0
+    fi
   fi
-fi
 
-if [ $${installation_needed} -eq 1 ]; then
-  # Download and install Vault
-  if [ -z "$${CURRENT_VERSION}" ]; then
-    printf "Installing Vault CLI ...\n\n"
-  else
-    printf "Upgrading Vault CLI from version %s to %s ...\n\n" "$${CURRENT_VERSION}" "$${VERSION}"
-  fi
-  fetch vault.zip "https://releases.hashicorp.com/vault/$${VERSION}/vault_$${VERSION}_linux_amd64.zip"
-  if [ $? -ne 0 ]; then
-    printf "Failed to download Vault.\n"
-    exit 1
-  fi
-  unzip vault.zip
-  if [ $? -ne 0 ]; then
-    printf "Failed to unzip Vault.\n"
-    exit 1
-  fi
-  rm vault.zip
-  if sudo mv vault /usr/local/bin/vault 2> /dev/null; then
-    printf "Vault installed successfully!\n\n"
-  else
-    mkdir -p ~/.local/bin
-    mv vault ~/.local/bin/vault
-    if [ ! -f ~/.local/bin/vault ]; then
-      printf "Failed to move Vault to local bin.\n"
-      exit 1
+  if [ $${installation_needed} -eq 1 ]; then
+    # Download and install Vault
+    if [ -z "$${CURRENT_VERSION}" ]; then
+      printf "Installing Vault CLI ...\n\n"
+    else
+      printf "Upgrading Vault CLI from version %s to %s ...\n\n" "$${CURRENT_VERSION}" "${INSTALL_VERSION}"
+    fi
+    fetch vault.zip "https://releases.hashicorp.com/vault/$${INSTALL_VERSION}/vault_$${INSTALL_VERSION}_linux_amd64.zip"
+    if [ $? -ne 0 ]; then
+      printf "Failed to download Vault.\n"
+      return 1
+    fi
+    if ! unzip_safe vault.zip; then
+      printf "Failed to unzip Vault.\n"
+      return 1
+    fi
+    rm vault.zip
+    if sudo mv vault /usr/local/bin/vault 2> /dev/null; then
+      printf "Vault installed successfully!\n\n"
+    else
+      mkdir -p ~/.local/bin
+      if ! mv vault ~/.local/bin/vault; then
+        printf "Failed to move Vault to local bin.\n"
+        return 1
+      fi
+      printf "Please add ~/.local/bin to your PATH to use vault CLI.\n"
     fi
-    printf "Please add ~/.local/bin to your PATH to use vault CLI.\n"
   fi
+  return 0
+}
+
+TMP=$(mktemp -d)
+if ! (
+  cd $TMP
```

Comment:
It may be rare, but tmpdir could contain spaces
```suggestion
  cd "$TMP"
```